backend/tracim_backend/applications/agenda/lib.py

```diff
@@ -233,7 +247,51 @@ def get_user_addressbook_url(self, user: User, use_proxy: bool) -> str:
             base_url=base_url, user_agenda_path=user_agenda_path
         )
 
-    def ensure_workspace_agenda_exists(self, workspace: Workspace) -> bool:
+    def sync_user_agenda(self, user: User) -> AgendaSyncState:
+        if self.user_has_agenda_enabled(user):
+            if self._ensure_user_agenda_exists(user=user):
+                state = AgendaSyncState.EXISTING
+            else:
+                state = AgendaSyncState.CREATED
+        else:
+            # INFO - GM - 2022-27-01  - We do not delete existing agenda if set to disabled
+            state = AgendaSyncState.DISABLED
+        self.sync_user_symlinks(user=user)
+        return state
+
+    def sync_workspace_agenda(self, workspace: Workspace) -> AgendaSyncState:
+        if self.workspace_has_agenda_enabled(workspace):
+            if self._ensure_workspace_agenda_exists(workspace=workspace):
+                state = AgendaSyncState.EXISTING
+            else:
+                state = AgendaSyncState.CREATED
+        else:
+            # INFO - GM - 2022-27-01  - We do not delete existing agenda if set to disabled
+            state = AgendaSyncState.DISABLED
+        for role in workspace.roles:
+            self.sync_workspace_symlinks(workspace=workspace, user=role.user)
+        return state
+
+    def sync_user_symlinks(self, user: User):
+        if self.user_has_agenda_enabled(user):
+            self._ensure_user_symlinks_exist(user)
+        else:
+            self._ensure_user_symlinks_missing(user)
+
+    def sync_workspace_symlinks(self, user: User, workspace: Workspace, role_deletion=False):
+        # TODO - G.M - 2022-01-27 - Reconsider role_deletion option with change in role before
+        # deletion like deleted status for User or Workspace ?
```

Comment:
I use a role_deletion parameter here as there is not other way to pass the deletion information without risking breaking something. I have considered using the NOT_APPLICABLE workspace role (value 0), but this change the received TLM for deletion and i'm not sure how well this role is supported in api and frontend.

@raphj , are you ok with this ?